### PR TITLE
feat: update TeleportedTetherToken contract

### PR DESCRIPTION
**Pull request by imjacklai**
Ticket: https://app.asana.com/0/1206918845779480/1208051219734858/f

- Update contract to Cadence 1.0
- Add related transactions
- Add related scripts
- Write test cases

### CODE REVIEW

**Review by andrew54068 — APPROVED** [review]

File: contracts/flow/token/TeleportedTetherToken.cdc [hunk 1]
```diff
@@ -104,101 +154,158 @@ pub contract TeleportedTetherToken: FungibleToken {
...
-    pub var balance: UFix64
+  // Gets a list of the metadata views that this contract supports
+  access(all) view fun getContractViews(resourceType: Type?): [Type] {
+    return [Type<FungibleTokenMetadataViews.FTView>(),
+            Type<FungibleTokenMetadataViews.FTDisplay>(),
+            Type<FungibleTokenMetadataViews.FTVaultData>(),
+            Type<FungibleTokenMetadataViews.TotalSupply>()]
+  }
+
+  // Get a Metadata View from FlowToken
```
Comment (joshuahannan) [inline contracts/flow/token/TeleportedTetherToken.cdc:169]: FlowToken?

**Review by joshuahannan — APPROVED** [review]
Everything looks pretty good!

**Review by scottphc — APPROVED** [review]

File: contracts/flow/token/TeleportedTetherToken.cdc (hunk 1 above)
Comment (imjacklai) [reply contracts/flow/token/TeleportedTetherToken.cdc]: Fixed, thanks for your review.

**Comment by boczeratul** [comment]
@imjacklai is this on testnet already?

**Comment by boczeratul** [comment]
Ah saw it https://testnet.flowdiver.io/contract/A.2d270db9ac8c7fef.TeleportedTetherToken?tab=deployments

File: contracts/flow/token/TeleportedTetherToken.cdc [hunk 2]
```diff
@@ -104,101 +154,158 @@ pub contract TeleportedTetherToken: FungibleToken {
...
+          symbol: "tUSDT",
+          description: "tUSDT stands for teleported USDT on Flow Blockchain.",
+          externalURL: MetadataViews.ExternalURL("https://blocto.io"),
+          logos: medias,
+          socials: {
+            "twitter": MetadataViews.ExternalURL("https://twitter.com/BloctoApp")
+          }
+        )
+      case Type<FungibleTokenMetadataViews.FTVaultData>():
+        let vaultRef = TeleportedTetherToken.account.storage.borrow<auth(FungibleToken.Withdraw) &TeleportedTetherToken.Vault>(from: /storage/TeleportedTetherTokenVault)
```
Comment (boczeratul) [inline contracts/flow/token/TeleportedTetherToken.cdc:200]: `(from: /storage/teleportedTetherTokenVault)`

**Review by boczeratul — CHANGES_REQUESTED** [review]

File: contracts/flow/token/TeleportedTetherToken.cdc (hunk 2 above)
Comment (boczeratul) [reply contracts/flow/token/TeleportedTetherToken.cdc:200]: typo

Comment (boczeratul) [reply contracts/flow/token/TeleportedTetherToken.cdc:200]: Also in the contract init method, you did not create the vault
```
self.account.storage.save(<-vault, to: /storage/teleportedTetherTokenVault)
```
Here you will run into 
```
panic("Could not borrow reference to the contract's Vault!")
```

Comment (imjacklai) [reply contracts/flow/token/TeleportedTetherToken.cdc:200]: Is it necessary to create the vault in the `init` method? As I know, the contract owner (administrator) might not need the vault.
If it’s needed, a transaction can be sent later to create the vault.

Comment (imjacklai) [reply contracts/flow/token/TeleportedTetherToken.cdc:200]: Got your point, will fix it.

Comment (boczeratul) [reply contracts/flow/token/TeleportedTetherToken.cdc:200]: I sent that transaction on testnet to initiate the tUSDT token vault on this contract BTW :P
So I can work on the tUSDT swap functionalities.

Comment (imjacklai) [reply contracts/flow/token/TeleportedTetherToken.cdc]: fixed.

**Review by boczeratul — APPROVED** [review]

**Pull request merged**